From d319cf5e756a96efca093c28d645e8b469233f0f Mon Sep 17 00:00:00 2001
From: Alex Gonzalez <alex.gonzalez@digi.com>
Date: Mon, 18 Mar 2019 18:45:13 +0100
Subject: [PATCH] ccimx6ul: Add BalenaOS partition table

BalenaOS requires the following partitioning:

256MB variant:
mtdparts=gpmi-nand:3m(bootloader),3m(environment),20m(resin-state),24m(resin-boot),100m(resin-rootA),100m(resin-rootB),-(resin-data)

1024Mb variant:
mtdparts=gpmi-nand:3m(bootloader),3m(environment),20m(resin-state),24m(resin-boot),312m(resin-rootA),312m(resin-rootB),-(resin-data)

Signed-off-by: Alex Gonzalez <alex.gonzalez@digi.com>
---
 include/configs/ccimx6ul_common.h | 26 +++++++++++++++-----------
 include/configs/ccimx6ulsbc.h     |  7 +++++--
 include/configs/ccimx6ulstarter.h |  7 +++++--
 scripts/config_whitelist.txt      |  4 ++++
 4 files changed, 29 insertions(+), 15 deletions(-)

diff --git a/include/configs/ccimx6ul_common.h b/include/configs/ccimx6ul_common.h
index ba4de9b43588..540c9a7a1f8b 100644
--- a/include/configs/ccimx6ul_common.h
+++ b/include/configs/ccimx6ul_common.h
@@ -188,26 +188,30 @@
 /* MTD (NAND) */
 #define CONFIG_SKIP_NAND_BBT_SCAN
 #define CONFIG_UBOOT_PARTITION		"bootloader"
-#define CONFIG_LINUX_PARTITION		"linux"
-#define CONFIG_RECOVERY_PARTITION	"recovery"
+#define CONFIG_LINUX_PARTITION		"resin-boot"
+#define CONFIG_ROOTFS_PARTITION 	"resin-rootA"
+#define CONFIG_RECOVERY_PARTITION	"resin-rootB"
+#define CONFIG_SAFE_PARTITION		"resin-state"
+#define CONFIG_UPDATE_PARTITION 	"resin-data"
 #define CONFIG_NAND_NAME		"gpmi-nand"
 #define MTDIDS_DEFAULT			"nand0=" CONFIG_NAND_NAME
 #define MTDPARTS_256MB			"mtdparts=" CONFIG_NAND_NAME ":" \
 					"3m(" CONFIG_UBOOT_PARTITION ")," \
 					"1m(environment)," \
-					"1m(safe)," \
+					"10m(" CONFIG_SAFE_PARTITION ")," \
 					"12m(" CONFIG_LINUX_PARTITION ")," \
-					"14m(" CONFIG_RECOVERY_PARTITION ")," \
-					"122m(rootfs)," \
-					"-(update)"
+					"100m(" CONFIG_RECOVERY_PARTITION ")," \
+					"100m(" CONFIG_ROOTFS_PARTITION ")," \
+					"-(" CONFIG_UPDATE_PARTITION ")"
 #define MTDPARTS_1024MB			"mtdparts=" CONFIG_NAND_NAME ":" \
 					"3m(" CONFIG_UBOOT_PARTITION ")," \
 					"3m(environment)," \
-					"1m(safe)," \
-					"24m(" CONFIG_LINUX_PARTITION ")," \
-					"32m(" CONFIG_RECOVERY_PARTITION ")," \
-					"512m(rootfs)," \
-					"-(update)"
+					"20m(" CONFIG_SAFE_PARTITION ")," \
+					"12m(" CONFIG_LINUX_PARTITION ")," \
+					"312m(" CONFIG_RECOVERY_PARTITION ")," \
+					"312m(" CONFIG_ROOTFS_PARTITION ")," \
+					"-(" CONFIG_UPDATE_PARTITION ")"
+#define CONFIG_ENV_MTD_SAFE_INDEX	"2"
 #define CONFIG_ENV_MTD_LINUX_INDEX	"3"
 #define CONFIG_ENV_MTD_RECOVERY_INDEX	"4"
 #define CONFIG_ENV_MTD_ROOTFS_INDEX	"5"
diff --git a/include/configs/ccimx6ulsbc.h b/include/configs/ccimx6ulsbc.h
index 19040b0ea78b..8a5b45437297 100644
--- a/include/configs/ccimx6ulsbc.h
+++ b/include/configs/ccimx6ulsbc.h
@@ -185,8 +185,10 @@
 	CONFIG_ENV_MTD_SETTINGS \
 	"bootargs_linux=\0" \
 	"bootargs_nand_linux=setenv bootargs console=${console},${baudrate} " \
-		"${bootargs_linux} ${mtdparts} ubi.mtd=${mtdlinuxindex} " \
-		"ubi.mtd=${mtdrootfsindex} root=ubi1_0 " \
+		"${bootargs_linux} ${mtdparts} ubi.mtd=${mtdsafeindex} " \
+		"ubi.mtd=${mtdlinuxindex} ubi.mtd=${mtdrecoveryindex} " \
+		"ubi.mtd=${mtdrootfsindex} ubi.mtd=${mtdupdateindex} " \
+		"root=ubi3_0 " \
 		"rootfstype=ubifs rw " \
 		"${bootargs_once} ${extra_bootargs}\0" \
 	"install_linux_fw_sd=if load mmc 0 ${loadaddr} install_linux_fw_sd.scr;then " \
@@ -203,6 +205,7 @@
 			"fi;" \
 		"fi;\0" \
 	"mtdbootpart=" CONFIG_LINUX_PARTITION "\0" \
+	"mtdsafeindex=" CONFIG_ENV_MTD_SAFE_INDEX "\0" \
 	"mtdlinuxindex=" CONFIG_ENV_MTD_LINUX_INDEX "\0" \
 	"mtdrecoveryindex=" CONFIG_ENV_MTD_RECOVERY_INDEX "\0" \
 	"mtdrootfsindex=" CONFIG_ENV_MTD_ROOTFS_INDEX "\0" \
diff --git a/include/configs/ccimx6ulstarter.h b/include/configs/ccimx6ulstarter.h
index 5f23f5cce4b2..0a1f85d8d86b 100644
--- a/include/configs/ccimx6ulstarter.h
+++ b/include/configs/ccimx6ulstarter.h
@@ -185,8 +185,10 @@
 	CONFIG_ENV_MTD_SETTINGS \
 	"bootargs_linux=\0" \
 	"bootargs_nand_linux=setenv bootargs console=${console},${baudrate} " \
-		"${bootargs_linux} ${mtdparts} ubi.mtd=${mtdlinuxindex} " \
-		"ubi.mtd=${mtdrootfsindex} root=ubi1_0 " \
+		"${bootargs_linux} ${mtdparts} ubi.mtd=${mtdsafeindex} " \
+		"ubi.mtd=${mtdlinuxindex} ubi.mtd=${mtdrecoveryindex} " \
+		"ubi.mtd=${mtdrootfsindex} ubi.mtd=${mtdupdateindex} " \
+		"root=ubi3_0 " \
 		"rootfstype=ubifs rw " \
 		"${bootargs_once} ${extra_bootargs}\0" \
 	"install_linux_fw_sd=if load mmc 0 ${loadaddr} install_linux_fw_sd.scr;then " \
@@ -203,6 +205,7 @@
 			"fi;" \
 		"fi;\0" \
 	"mtdbootpart=" CONFIG_LINUX_PARTITION "\0" \
+	"mtdsafeindex=" CONFIG_ENV_MTD_SAFE_INDEX "\0" \
 	"mtdlinuxindex=" CONFIG_ENV_MTD_LINUX_INDEX "\0" \
 	"mtdrecoveryindex=" CONFIG_ENV_MTD_RECOVERY_INDEX "\0" \
 	"mtdrootfsindex=" CONFIG_ENV_MTD_ROOTFS_INDEX "\0" \
diff --git a/scripts/config_whitelist.txt b/scripts/config_whitelist.txt
index 111bc05a2362..f5f2505119f7 100644
--- a/scripts/config_whitelist.txt
+++ b/scripts/config_whitelist.txt
@@ -997,6 +997,7 @@ CONFIG_ENV_MIN_ENTRIES
 CONFIG_ENV_MTD_LINUX_INDEX
 CONFIG_ENV_MTD_RECOVERY_INDEX
 CONFIG_ENV_MTD_ROOTFS_INDEX
+CONFIG_ENV_MTD_SAFE_INDEX
 CONFIG_ENV_MTD_SETTINGS
 CONFIG_ENV_MTD_UPDATE_INDEX
 CONFIG_ENV_OFFSET
@@ -2636,6 +2637,7 @@ CONFIG_ROCKCHIP_SDHCI_MAX_FREQ
 CONFIG_ROCKCHIP_USB2_PHY
 CONFIG_ROM_STUBS
 CONFIG_ROOTFS_OFFSET
+CONFIG_ROOTFS_PARTITION
 CONFIG_ROOTPATH
 CONFIG_RSK7203
 CONFIG_RSK7264
@@ -2692,6 +2694,7 @@ CONFIG_S5P_PA_SYSRAM
 CONFIG_S6E63D6
 CONFIG_S6E8AX0
 CONFIG_SABRELITE
+CONFIG_SAFE_PARTITION
 CONFIG_SAMA5D2
 CONFIG_SAMA5D3
 CONFIG_SAMA5D3_LCD_BASE
@@ -6697,6 +6700,7 @@ CONFIG_UNIPHIER_ETH
 CONFIG_UPDATEB
 CONFIG_UPDATEFILE_SUPPORTED_SOURCES_ARGS_HELP
 CONFIG_UPDATE_LOAD_ADDR
+CONFIG_UPDATE_PARTITION
 CONFIG_UPDATE_SUPPORTED_SOURCES_ARGS_HELP
 CONFIG_UPDATE_SUPPORTED_SOURCES_LIST
 CONFIG_UPDATE_TFTP
